File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        char c = s.charAt(0);
        int same = 0;
        int diff = 0;
        int answer = 0;
        
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == c) same++;
            else diff++;
            
            if (same == diff) {
                answer++;
                if (i + 1 < s.length()) {
                    c = s.charAt(i + 1);
                    same = 0;
                    diff = 0;
                }
            }
        }
        
        if (same != diff) answer++;

        return answer;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.kt
class Solution {
    fun solution(s: String): Int {
        var same = 0
        var diff = 0
        var answer = 0
        var curChar = s[0]
        
        for ((i, c) in s.withIndex()) {
            if (c == curChar) same++
            else diff++
            
            if (same == diff) {
                answer++
                if (i + 1 < s.length) {
                    curChar = s[i + 1]
                    same = 0
                    diff = 0
                }
            }
        }
        
        if (same != diff) answer++
        
        return answer
    }
}